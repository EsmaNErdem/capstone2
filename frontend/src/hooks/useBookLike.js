import { useState, useEffect, useContext } from 'react';
import UserContext from '../auth/UserContext';

/** Custom Hook for managing user book like
 * This hook provides functionality to manage user likes on books, including retrieving initial like status and count,
 * updating the like status and count, and handling user interactions to like or unlike a book.
 * 
 * -Utilizes useContext data to send user like to database
 * -Creates state to update book like count and user book like status
 * -Use useEffect to avoid to recalculations
 * 
 * Returns user like status, book like count states and handleLike function
 */
const useBookLike = (bookId, initialLikes, bookData={}) => {
    console.debug("useBookLike"); 

    const { hasLikedBook, likeBook } = useContext(UserContext);
    const [liked, setLiked] = useState();
    const [likes, setLikes] = useState();
    const [error, setError] = useState(null);

    // Helper function to convert "undefined" or "null" strings to actual undefined
    const convertUndefined = (value) => (value === "undefined" || value === null ? undefined : value);

    // Convert undefined/null strings in the bookData object
    const convertedBookData = Object.fromEntries(
        Object.keys(bookData).map((key) => [key, convertUndefined(bookData[key])])
    );
        
    /**By using the useEffect, the liked status is only recalculated when the id avoiding unnecessary recalculations on every render.  */
    useEffect(() => {
      setLiked(hasLikedBook(bookId));
      setLikes(initialLikes)
    }, [bookId, initialLikes]);
    
    /**
     * Send user like to database
     * Updates the like count and status based on user actions.
     */ 
    const handleLikeBook = async () => {
        try {
            setError(null);
            if (liked) {
                const unlikedBookId = await likeBook(bookId);
                if (unlikedBookId === bookId) {
                    setLikes((likes) => likes - 1);
                    setLiked(false);
                }
            } else {
                const likedBookId = await likeBook(bookId, convertedBookData);
                if (likedBookId === bookId) {
                    setLikes((likes) => likes + 1);
                    setLiked(true);
                }
            }
        } catch (error) {
            setError('Error liking book.');
            console.error('Error handling book like:', error);
        }
    };
    
  return { liked, likes, error, handleLikeBook };
};

export default useBookLike;