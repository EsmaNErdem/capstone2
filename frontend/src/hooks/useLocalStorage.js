import { useState, useEffect } from "react";

/**
 * Custom Hook to save user info like token and username into localStorage
 * 
 * Creates a user state and looks in localStorage for current user info, if cannot find it, assign null
 * When user info updates, useEffect either cleans it or updates localStorage
 * 
 * Retunrs [user, setUser]
 */

const useLocalStorage = (localStorageKey, initialValue = null) => {
    let initialUserValue
    if(localStorage.getItem(localStorageKey)) {
        try {
            initialUserValue = JSON.parse(localStorage.getItem(localStorageKey))
        } catch (e) {
            console.error("useLocalStorage custom hook: problem loading localStorage data", e);
        }
    } else {
        initialUserValue = initialValue
    }

    const [user, setUser] = useState(initialUserValue)

    useEffect(() => {
        if(user === null) {
            localStorage.removeItem(localStorageKey)
        } else {
            localStorage.setItem(localStorageKey, JSON.stringify(user))
        }
    }, [localStorageKey, user])

    return [user, setUser]
}

export default useLocalStorage;