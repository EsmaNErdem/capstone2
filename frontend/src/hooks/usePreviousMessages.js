import { useState, useEffect } from "react";
import BookClubApi from "../api";

/** Custom Hook for fetching previous messages between receiver and current user
 * This hook makes API call to backend to retrieve related data and set messaegs state
 * 
 * Returns error and messages
 */
const usePreviousMessages = (receiver, currentUserName, websocket, isOpen) => {
    const [error, setError] = useState(null);
    const [messages, setMessages] = useState([]);

    /**
     * Fetches the previous messages between users from backend sends to child component after WebSocket connection is established
     */ 
    useEffect(function getPreviousMessages() {
        console.debug("usePreviousMessages hook useEffect getPreviousMessages");
    
        const getMessages = async () => {
            try{
                if (isOpen && websocket) {
                    const roomName = `${receiver},${currentUserName}`
                    const previousMessages = await BookClubApi.getRoomPreviousMessages(roomName);
                    setMessages([...previousMessages]);                 
                }
            } catch (e) {
                console.error("usePreviousMessages useEffect API call data loading error:", e);
                setError("An error occurred while fetching previous messages.");
            }
        }
        getMessages();
    }, [isOpen, receiver, websocket]);

    return { error, messages }
}

export default usePreviousMessages;