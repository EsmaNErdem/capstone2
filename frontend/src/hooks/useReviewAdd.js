import { useState, useContext } from 'react';
import UserContext from '../auth/UserContext';

/** Custom Hook for managing user review submission on books/:id route components
 * This hook provides functionality to manage user review submission and updateting bookReviews state
 * 
 * -Utilizes useContext data to send user review to database
 * -Updates book review
 * 
 * Returns error and addBookReview function
 */
const useReviewAdd = (setBookReviews, bookData={}, errorHandlingState) => {
    console.debug("useReviewAdd"); 

    const { addUserReview } = useContext(UserContext);
    const [error, setError] = useState(null);

    // add current user's book review, function is called in BookReviewDrawer
    const addBookReview = async ({ review }) => {
        try {
            console.log({book: bookData, review})
            const newReview = await addUserReview({book: bookData, review});
            setBookReviews(r => [{...newReview, id:bookData.id}, ...r])
            return newReview
        } catch (error) {
            errorHandlingState(false)
            setError("Error adding book review.")
            console.error("Error adding book review:", error);
        }
    } 
    
    return { error, addBookReview }

}

export default useReviewAdd