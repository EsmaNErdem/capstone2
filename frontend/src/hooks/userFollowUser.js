import { useState, useEffect, useContext } from 'react';
import UserContext from '../auth/UserContext';

/** Custom Hook for managing user follow
 * This hook provides functionality to manage user follow on user profile page, retrieves initial follow status,
 * updating the follow status and user followers list, and handling user follow data to database
 * 
 * -Utilizes useContext data to send user like to database
 * -Creates state to update user follow status
 * -Use useEffect to avoid to recalculations
 * 
 * Returns user follow status, related error nand handleFollowUser function
 */
const useFollowUser = (userToBeFollowed, setUserFollow, currUserProfile, userCard, userToBeFollowedImg=null) => {
    console.debug("useFollowUser"); 
    const { hasFollowing, followUser } = useContext(UserContext);
    
    const [followed, setFollowed] = useState()
    const [error, setError] = useState(null);
    
    /**By using the useEffect, the followed status is only recalculated when username changes  */
    useEffect(() => {
        console.debug("Follow user useEffect, userToBeFollowed=", userToBeFollowed)

        setFollowed(hasFollowing(userToBeFollowed))
    }, [userToBeFollowed])
    
    /** 
     * Sends user follow and unfollow data to database
     * -Updates users followers list if user on profile view is not current user and handleFollow is called from profile follow button
     * -Updates users following list if user on profile view is current user and handleFollow is called from  user profile card follow button
    */
    const handleFollowUser = async () =>{
        try {
            setError(null);
            if (followed) {
                const unfollowedUsername = await followUser(userToBeFollowed);
                if (!currUserProfile && !userCard) {
                    // Current user is not on the profile, and the unfollow button is not from the user profile card
                    setUserFollow(followers => followers.filter(follower => follower.followedBy !== unfollowedUsername));
                } else if (currUserProfile && userCard) {
                    // Current user is on the profile, and the unfollow button is from the user profile card
                    setUserFollow(followings => followings.filter(following => following.following !== userToBeFollowed));
                } 

                setFollowed(false) 
            } else {
                const followingUser = await followUser(userToBeFollowed);
                if (!currUserProfile && !userCard) {
                    // Current user is not on the profile, and the follow button is not from the user profile card
                    setUserFollow(followers => [...followers, followingUser]);
                } else if (currUserProfile && userCard) {
                    // Current user is on the profile, and the follow button is from the user profile card
                    setUserFollow(followings => [...followings, { following: userToBeFollowed, userImg: userToBeFollowedImg }]);
                }                  

                setFollowed(true) 
            }
        } catch (error) {
            setError(`Error following user: ${error}`)
            console.error("Error handling user follow:", error);
        }
    }
    
    return { followed, error, handleFollowUser }
}

export default useFollowUser;