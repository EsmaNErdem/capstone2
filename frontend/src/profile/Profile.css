.Profile {
    width: 100%;
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: stretch;
    padding: 2% 15%;
}

.selectedTab {
    font-weight: bold;
    color: #a074f3;
  }
  
.appBarWithShadow {
    border: 1px solid yellowgreen;
    box-shadow: 0px 4px 6px rgba(17, 192, 44, 0.1);
}

.Profile-LikeBooks-List { 
    display: flex;
    flex-wrap: wrap;
    justify-content: space-around;
    margin-top: 20px; 
}
