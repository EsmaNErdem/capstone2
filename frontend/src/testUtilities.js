import React from "react";
import UserContext from "./auth/UserContext";

const demoUser = {
  username: "testuser",
  firstNsme: "testfirst",
  lastName: "testlast",
  email: "test@test.net",
  img: null,
};

const UserProvider =
    ({ children, currentUser = demoUser, hasLikedReview = () => false, hasLikedBook = () => false, isUserReview = () => false, hasFollowing = () => false, setCurrentUser = u => u }) => (
    <UserContext.Provider value={{ currentUser, hasLikedReview, hasLikedBook, hasFollowing, setCurrentUser, isUserReview }}>
      {children}
    </UserContext.Provider>
);

const CurrUserProvider =
    ({ children, currentUser = demoUser, hasLikedReview = () => false, hasLikedBook = () => false, isUserReview = () => true, hasFollowing = () => false, setCurrentUser = u => u }) => (
    <UserContext.Provider value={{ currentUser, hasLikedReview, hasLikedBook, hasFollowing, setCurrentUser, isUserReview }}>
      {children}
    </UserContext.Provider>
);

const NonUserProvider =
    ({ children, currentUser = null, hasLikedReview = () => false, hasLikedBook = () => false, hasFollowing = () => false }) => (
    <UserContext.Provider value={{ currentUser, hasLikedReview, hasLikedBook, hasFollowing,}}>
      {children}
    </UserContext.Provider>
);

export { UserProvider, CurrUserProvider, NonUserProvider };
