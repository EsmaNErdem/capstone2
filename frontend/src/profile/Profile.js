import React, { useContext, useState, useEffect } from "react";
import { useParams } from "react-router-dom";
import useReviewDelete from "../hooks/useReviewDelete";
import BookClubApi from "../api";
import ReviewDisplay from "../reviews/ReviewDisplay";
import BookCard from "../books/BookCard";
import ProfileCard from "./ProfileCard";
import ProfileEditForm from "./ProfileEditForm"
import Loading from "../utilities/Loading";
import Alert from "../utilities/Alert";
import UserContext from '../auth/UserContext';
import useFollowUser from "../hooks/userFollowUser";
import { AppBar, Avatar, Box, Tab, Tabs, Typography, Modal, IconButton  } from '@mui/material';
import EditIcon from '@mui/icons-material/Edit';
import "./Profile.css"

/**Display User Profile data
 * Display list user related data under tabs and check user on view is whether current user and changes view accordingly.
 * 
 * - Shows list of reviews owned by user, reviews and books liked by user, and user's following and followers. 
 * - Displays edit profile form in modal if user is current user
 * - Handle user follow if user is not current user
 * 
 * - Routes ==> BookCard, ReviewDisplay, ProfileEditForm
 */
const Profile = () => {
    console.debug("Profile");

    const { username } = useParams();
    const { currentUser } = useContext(UserContext);

    // checks if user on view is current user
    const currUserProfile = currentUser.username === username;

    const [loading, setLoading] = useState(true);
    const [value, setValue] = useState(0);
    const [error, setError] = useState(null);
    const [user, setUser] = useState();
    const [userFollowings, setUserFollowings] = useState([]);
    const [userFollowers, setUserFollowers] = useState([]);
    const [userReviews, setUserReviews] = useState([]);
    const [userLikedReviews, setUserLikedReviews] = useState([]);
    const [userLikedBooks, setUserLikedBooks] = useState([]);
    const [userEditFormOpen, setUserEditFormOpen] = useState(false);

    const { error: deleteError, deleteBookReview } = useReviewDelete(setUserReviews);
    const { followed, error: followUser, handleFollowUser } = useFollowUser(user?.username, setUserFollowers, currUserProfile, false)

    /**
     * Fetches the user data when component mounts
     */  
    useEffect(function getProfileOnMount() {
        console.debug("Profile useEffect getProfileOnMount");
        
        // Make an API call to get user profile data
        const getuserProfile =  async () => {
        setError(null);
        try {
            const userApi = await BookClubApi.getUser(username);
            setUser(userApi);
            setUserFollowings(userApi.following)
            setUserFollowers(userApi.followers)
            setUserReviews(userApi.reviews)
            setUserLikedReviews(userApi.likedReviews)
            setUserLikedBooks(userApi.likedBooks)
        } catch (e) {
            console.error("Profile useEffect API call data loading error:", e);
            setError("An error occurred while fetching user profile.");
        }
        
        setLoading(false)
        }

        // set loading to true while async getCurrentUser runs; once the
        // data is fetched (or even if an error happens!), this will be set back
        // to true to control the spinner.
        setLoading(true)
        getuserProfile();
    }, [username]);

    // Opens Modal to display user profile edit form
    const openEditProfileForm = async () => {
        setUserEditFormOpen(true)
    }

    // Closes Modal to display user profile edit form
    const closeEditProfileForm = async () => {
        setUserEditFormOpen(false)
    }

    if(!user || loading) return <Loading />;

    return (
        <div className="Profile">
            <Box sx={{ display: 'flex', alignItems: 'center', padding: '1rem', marginBottom: '2rem' }}>
                <Avatar alt={user.username} src={user.img} sx={{ width: 60, height: 60 }} />
                <Box sx={{ display: 'flex', flexDirection: 'column', marginLeft: '1rem' }}>
                    <Box sx={{ display: 'flex', alignItems: 'center', justifyContent: "space-between"}}>
                        <Typography variant="h4">{user.username}</Typography>
                        { currUserProfile ? (
                            <IconButton 
                                onClick={openEditProfileForm}
                                color="primary" 
                                aria-label="edit" 
                                sx={{ marginLeft: '2rem', color:"orangered" }}
                                data-testid="edit-profile-button"
                            >
                                <EditIcon /><span>Edit</span>
                            </IconButton>
                            )
                            : (
                                <IconButton 
                                    onClick={handleFollowUser} 
                                    color="primary" 
                                    aria-label="follow" 
                                    sx={{ marginLeft: '2rem', color: '#6d17b7' }}
                                    data-testid="follow-profile-button"
                                >
                                    {followed ? "Unfollow" : "+ Follow"}
                                </IconButton>
                            )
                        }
                    </Box>
                    {currUserProfile &&
                        <Box sx={{ display: 'flex', justifyContent: 'flex-start', alignItems: 'center', marginLeft: '1rem' }}>
                            <Typography variant="h6">{user.firstName} {user.lastName}</Typography>
                        </Box>
                    }
                </Box>
            </Box> 

            <Modal open={userEditFormOpen} onClose={closeEditProfileForm}>
                <Box className="Profile-edit" >
                    <ProfileEditForm 
                        close={true} 
                        closeModal={closeEditProfileForm}
                        initialValues={{
                            username: user.username,
                            firstName: user.firstName,
                            lastName: user.lastName,
                            img: user.img,
                            email: user.email,
                        }}
                        updateUser={setUser}
                        />
                </Box>
            </Modal>

            {error || deleteError || followUser ? <Alert type="danger" messages={[error || deleteError || followUser]} />: null}

            <AppBar position="static" color="transparent" className="appBarWithShadow">
                <Tabs value={value} onChange={(event, newValue) => setValue(newValue)} >
                    <Tab label={`${userReviews.length} User Reviews`} className={value === 0 ? "selectedTab" : ""} />
                    <Tab label={`${userLikedReviews.length} Liked Reviews`} className={value === 1 ? "selectedTab" : ""} />
                    <Tab label={`${userLikedBooks.length} Liked Books`} className={value === 2 ? "selectedTab" : ""} />
                    <Tab label={`${userFollowers.length} Followers`} className={value === 3 ? "selectedTab" : ""} />
                    <Tab label={`${userFollowings.length} Following`} className={value === 4 ? "selectedTab" : ""} />
                </Tabs>
            </AppBar>

            {value === 0 && (
                <div className="Profile-UserReviews-List">
                    {userReviews.map(({ reviewId, review, date, book_id, title, cover, author, category, reviewLikeCount }) => (
                      <ReviewDisplay 
                          key={reviewId}
                          reviewId={+reviewId}
                          review={review}
                          date={date}
                          username={user.username}
                          userImg={user.img}
                          bookId={book_id}
                          title={title}
                          author={author}
                          cover={cover}
                          category={category}
                          reviewLikeCount={+reviewLikeCount}
                          deleteReview={deleteBookReview}
                      />
                  ))}
                </div>
            )}
            {value === 1 && (
                <div className="Profile-LikedReviews-List">
                    {userLikedReviews.map(({ reviewId, review, date, username, userImg, book_id, title, cover, author, category, reviewLikeCount }) => (
                      <ReviewDisplay 
                          key={reviewId}
                          reviewId={+reviewId}
                          review={review}
                          date={date}
                          username={username}
                          userImg={userImg}
                          bookId={book_id}
                          title={title}
                          author={author}
                          cover={cover}
                          category={category}
                          reviewLikeCount={+reviewLikeCount}
                          likedReview={currUserProfile}
                          setReviews={setUserLikedReviews}
                      />
                    ))}                
                </div>
            )}
            {value === 2 && (
                <div className="Profile-LikeBooks-List">
                    {userLikedBooks.map(({ book_id, title, author, description, publisher, category, cover, bookLikeCount, reviews }) => (
                      <BookCard 
                          key={book_id}
                          id={book_id}
                          title={title}
                          author={author}
                          description={description}
                          publisher={publisher}
                          category={category}
                          cover={cover}
                          bookLikeCount={+bookLikeCount}
                          likedBook={true}
                          currUser={currUserProfile}
                          setBooks={setUserLikedBooks}
                      />
                  ))}
                </div>
            )}

            {value === 3 && (
                <div className="Profile-Followers-List">
                    {userFollowers.map((follower) => (
                        <ProfileCard
                            key={follower.followedBy}
                            username={follower.followedBy}
                            userImg={follower.userImg}
                            setUserFollow={setUserFollowings}
                            currUserProfile={currUserProfile}
                        />
                    ))}
                </div>
            )}

            {value === 4 && (
                <div className="Profile-Following-List">
                    {userFollowings.map((following) => (
                        <ProfileCard
                            key={following.following}
                            username={following.following}
                            userImg={following.userImg}
                            setUserFollow={setUserFollowings}
                            currUserProfile={currUserProfile}
                        />
                    ))}
                </div>
            )}

        </div>
    )
}

export default Profile;