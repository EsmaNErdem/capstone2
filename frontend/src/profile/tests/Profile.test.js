import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import { UserProvider } from "../../testUtilities";
import { render, fireEvent, act, waitFor } from '@testing-library/react';
import Profile from '../Profile';
import BookClubApi from '../../api';
import * as useFollowModule from '../../hooks/userFollowUser';

const mockUser = {
    username: "Test1",
    firstName: "TestFirs1",
    lastName: "TestLast1",
    img: "UserImg1",
    email: "u1@email.com",
    reviews: [{
        reviewId: 1,
        review: "Review 1",
        date: "12/18/2023",
        username: "Test1", 
        userImg: "UserImg1",
        book_id: 1,
        title: 'Book 1',
        cover: 'Cover 1',
        author: 'Author 1',
        category: 'Category 1',
        reviewLikeCount: 2
    }],
    likedBooks: [{
        book_id: '1',
        title: 'Book1',
        author: 'Author1',
        publisher: 'Publisher1',
        description: 'Description1',
        category: 'Category1',
        cover: 'Cover1',
        bookLikeCount: 5,
    }],
    likedReviews: [{
        reviewId: 2,
        review: "Review 2",
        date: "12/18/2023",
        username: "Test2", 
        userImg: "UserImg2",
        book_id: 1,
        title: 'Book 1',
        cover: 'Cover 1',
        author: 'Author 1',
        category: 'Category 1',
        reviewLikeCount: 2
    }],
    followers:[{
        followedBy: "Test2",
        userImg: 'UserImg2'
      }],
    following: [{
        following: 'Test2',
        userImg: 'UserImg2',
      }]
}

const mockFollow = {
    following: "Test1",
    followedBy: "Test4",
}

jest.mock("../../api", () => ({
    __esModule: true,
    default: {
        getUser: jest.fn(),
        followUser: jest.fn(),
    },
}));

beforeEach(() => {
    handleUserFollowMock = jest.fn();
  
    jest.spyOn(useFollowModule, 'default').mockImplementation(() => ({
      followed: false,
      error: null,
      handleFollowUser: handleUserFollowMock,
    }));
  });

afterEach(() => {
    jest.clearAllMocks();
});

test('renders without crashing', async () => {
    BookClubApi.getUser.mockImplementationOnce(() => (mockUser));
  
    await act(async () => {
      render(
        <MemoryRouter>
          <UserProvider>
            <Profile />
          </UserProvider>
        </MemoryRouter>
      );
    });
  
});

test('it renders and matches with snapshot', async () => {
    BookClubApi.getUser.mockImplementationOnce(() => (mockUser));
  
    await act(async () => {
      ({ asFragment, getByTestId } = render(
        <MemoryRouter>
          <UserProvider>
            <Profile />
          </UserProvider>
        </MemoryRouter>
      ));
    });
  
    expect(asFragment()).toMatchSnapshot();
});

test('renders mock data', async () => {
    BookClubApi.getUser.mockImplementationOnce(() => (mockUser));
    
    await act(async () => {
        ({ getByText, getByTestId, queryByText } = render(
        <MemoryRouter>
          <UserProvider>
            <Profile />
          </UserProvider>
        </MemoryRouter>
      ));
    });
    
    expect(getByText("1 User Reviews")).toBeInTheDocument()
    expect(getByText("+ Follow")).toBeInTheDocument()
    expect(queryByText("Edit")).not.toBeInTheDocument()
    expect(queryByText("1 User Reviews")).toBeInTheDocument()  
});

test('renders mock follow', async () => {
    BookClubApi.getUser.mockImplementationOnce(() => (mockUser));
    BookClubApi.followUser.mockImplementationOnce(() => (mockFollow));
    
    await act(async () => {
      ({ getByTestId } = render(
          <MemoryRouter>
          <UserProvider>
            <Profile />
          </UserProvider>
        </MemoryRouter>
      ));
    });

    const followUser = getByTestId("follow-profile-button")
    fireEvent.click(followUser);
    
    await waitFor(() => {
        expect(handleUserFollowMock).toHaveBeenCalled();
    });

});