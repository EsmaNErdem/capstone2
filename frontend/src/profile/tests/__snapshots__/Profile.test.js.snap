// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it renders and matches with snapshot 1`] = `
<DocumentFragment>
  <div
    class="Profile"
  >
    <div
      class="MuiBox-root css-1obp3ff"
    >
      <div
        class="MuiAvatar-root MuiAvatar-circular css-1lycc6k-MuiAvatar-root"
      >
        <img
          alt="Test1"
          class="MuiAvatar-img css-1pqm26d-MuiAvatar-img"
          src="UserImg1"
        />
      </div>
      <div
        class="MuiBox-root css-kra275"
      >
        <div
          class="MuiBox-root css-1lekzkb"
        >
          <h4
            class="MuiTypography-root MuiTypography-h4 css-5lbw0b-MuiTypography-root"
          >
            Test1
          </h4>
          <button
            aria-label="follow"
            class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-16nj3oi-MuiButtonBase-root-MuiIconButton-root"
            data-testid="follow-profile-button"
            tabindex="0"
            type="button"
          >
            + Follow
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
          <button
            aria-label="chat"
            class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-1d8rwkd-MuiButtonBase-root-MuiIconButton-root"
            data-testid="chat-button"
            tabindex="0"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="MessageIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M20 2H4c-1.1 0-1.99.9-1.99 2L2 22l4-4h14c1.1 0 2-.9 2-2V4c0-1.1-.9-2-2-2zm-2 12H6v-2h12v2zm0-3H6V9h12v2zm0-3H6V6h12v2z"
              />
            </svg>
             
            <span>
              Chat
            </span>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
    </div>
    <div
      class="PrivateSwipeArea-root css-x15wq9"
      style="width: 20px;"
    />
    <header
      class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorTransparent MuiAppBar-positionStatic appBarWithShadow css-v526sl-MuiPaper-root-MuiAppBar-root"
    >
      <div
        class="MuiTabs-root css-1ujnqem-MuiTabs-root"
      >
        <div
          class="MuiTabs-scroller MuiTabs-fixed css-jpln7h-MuiTabs-scroller"
          style="overflow: hidden; margin-bottom: 0px;"
        >
          <div
            class="MuiTabs-flexContainer css-heg063-MuiTabs-flexContainer"
            role="tablist"
          >
            <button
              aria-selected="true"
              class="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary Mui-selected selectedTab css-1h9z7r5-MuiButtonBase-root-MuiTab-root"
              role="tab"
              tabindex="0"
              type="button"
            >
              1 User Reviews
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
            <button
              aria-selected="false"
              class="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary css-1h9z7r5-MuiButtonBase-root-MuiTab-root"
              role="tab"
              tabindex="-1"
              type="button"
            >
              1 Liked Reviews
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
            <button
              aria-selected="false"
              class="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary css-1h9z7r5-MuiButtonBase-root-MuiTab-root"
              role="tab"
              tabindex="-1"
              type="button"
            >
              1 Liked Books
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
            <button
              aria-selected="false"
              class="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary css-1h9z7r5-MuiButtonBase-root-MuiTab-root"
              role="tab"
              tabindex="-1"
              type="button"
            >
              1 Followers
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
            <button
              aria-selected="false"
              class="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary css-1h9z7r5-MuiButtonBase-root-MuiTab-root"
              role="tab"
              tabindex="-1"
              type="button"
            >
              1 Following
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </div>
          <span
            class="MuiTabs-indicator css-1aquho2-MuiTabs-indicator"
            style="left: 0px; width: 0px;"
          />
        </div>
      </div>
    </header>
    <div
      class="Profile-UserReviews-List"
    >
      <li
        class="MuiListItem-root MuiListItem-gutters ReviewDisplay css-1nxmd3h-MuiListItem-root"
      >
        <div
          class="UserInfo"
        >
          <a
            class="UsernameLink"
            data-testid="review-display-profile-link"
            href="/profile/Test1"
          >
            <div
              class="MuiAvatar-root MuiAvatar-circular css-1fy7wd8-MuiAvatar-root"
            >
              <img
                alt="Test1"
                class="MuiAvatar-img css-1pqm26d-MuiAvatar-img"
                src="UserImg1"
              />
            </div>
            <div
              class="MuiListItemText-root MuiListItemText-multiline css-konndc-MuiListItemText-root"
            >
              <p
                class="MuiTypography-root MuiTypography-body2 css-e784if-MuiTypography-root"
                style="color: orangered; font-size: 1.2rem;"
              >
                Test1
              </p>
              <p
                class="MuiTypography-root MuiTypography-body2 css-e784if-MuiTypography-root"
                style="color: rgb(109, 23, 183);"
              >
                12/18/2023 12:00:00 AM
              </p>
            </div>
          </a>
          <div>
            <button
              class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
              data-testid="review-display-review-like"
              tabindex="0"
              type="button"
            >
              <span
                data-testid="display-like-count"
              >
                2
              </span>
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="FavoriteIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="m12 21.35-1.45-1.32C5.4 15.36 2 12.28 2 8.5 2 5.42 4.42 3 7.5 3c1.74 0 3.41.81 4.5 2.09C13.09 3.81 14.76 3 16.5 3 19.58 3 22 5.42 22 8.5c0 3.78-3.4 6.86-8.55 11.54L12 21.35z"
                />
              </svg>
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </div>
        </div>
        <div
          class="ReviewInfo"
        >
          <div
            class="MuiListItemText-root css-tlelie-MuiListItemText-root"
          >
            <span
              class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-1k48cxs-MuiTypography-root"
            >
              Review 1
            </span>
          </div>
        </div>
        <div
          class="BookInfo"
        >
          <a
            class="Title"
            href="/books/1"
          >
            <img
              alt="Book 1"
              class="CardImage"
              src="Cover 1"
            />
          </a>
          <div>
            <h2>
              <a
                class="TitleLink"
                data-testid="review-display-book-title-link"
                href="/books/1"
              >
                Book 1
              </a>
            </h2>
            <p>
              Author: Author 1
            </p>
            <p>
              Category: Category 1
            </p>
          </div>
        </div>
      </li>
      <hr
        class="MuiDivider-root MuiDivider-fullWidth css-9mgopn-MuiDivider-root"
      />
    </div>
  </div>
</DocumentFragment>
`;
